app/Models/Collection.php

```diff
@@ -577,4 +578,50 @@ public function scopeFeatured(Builder $query): Builder
     {
         return $query->where('is_featured', true);
     }
+
+    /**
+     * @param  Builder<self>  $query
+     * @return Builder<self>
+     */
+    public function scopeVotable(Builder $query): Builder
+    {
+        $subQuery = Collection::query()->select([
+            'collections.*',
+            DB::raw('COUNT(collection_votes.id) as votes_count'),
+            DB::raw('ROW_NUMBER() OVER (ORDER BY COUNT(collection_votes.id) DESC, volume DESC NULLS LAST) as rank'),
+        ])
+            ->leftJoin('collection_votes', function ($join) {
+                $join->on('collection_votes.collection_id', '=', 'collections.id')
+                    ->whereBetween('collection_votes.voted_at', [
+                        Carbon::now()->startOfMonth(),
+                        Carbon::now()->endOfMonth(),
+                    ]);
+            })
+            ->groupBy('collections.id');
```

Comment:
i'm curious, why is this query written this way? isn't this query basically the same as the other scopes (`whereHas`)?

Reply:
@crnkovic yes its the same but i was unable to use the collection scope because is part of a custom join query